app/app.go

```diff
@@ -25,12 +26,22 @@ var (
 	tpool *pool.Pool
```

Comment:
This definitely should not be global. The set of transactions a node/app instance knows about should not be shared by another instance just because they happen to be running in the same process. Pretty sure this should be a field on the app struct.